src/uct/ib/dc/dc_mlx5.c

```diff
@@ -1093,6 +1093,22 @@ void uct_dc_mlx5_fc_entry_iter_del(uct_dc_mlx5_iface_t *iface, khiter_t it)
     }
 }
 
+static UCS_F_ALWAYS_INLINE uct_dc_mlx5_ep_t*
+uct_dc_mlx5_fc_retrieve_ep(uct_dc_mlx5_iface_t *iface, uct_dc_mlx5_ep_t *ep,
+                           uct_dc_fc_sender_data_t *fc_data)
+{
+    khiter_t it = kh_get(uct_dc_mlx5_fc_hash, &iface->tx.fc_hash,
+                         (uint64_t)ep);
+    if ((it != kh_end(&iface->tx.fc_hash)) &&
+        ((fc_data == NULL) ||
```

Comment:
when can it be NULL? On line 1157 you unconditionally access this struct to get ep

Reply:
refactored